### PR TITLE
#38 - S3 storage support in YAML settings

### PR DESCRIPTION
**Pull request by olegmoz**
Part of issue #38 
Added support for S3 storage type to 'YamlSettings'

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#53](https://github.com/artipie/artipie/issues/53) is assigned to @vatavuk[/z](https://www.0crat.com/u/vatavuk), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/96da6aee-3aca-4f25-993a-14267a5dfb61); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/main/java/com/artipie/YamlSettings.java [hunk 1]
```diff
@@ -63,9 +70,51 @@ public Storage storage() throws IOException {
                 .yamlMapping("storage")
             );
         final String type =  yaml.string("type");
+        final Storage storage;
         if ("fs".equals(type)) {
-            return new FileStorage(Path.of(yaml.string("path")), Vertx.vertx().fileSystem());
+            storage = new FileStorage(Path.of(yaml.string("path")), Vertx.vertx().fileSystem());
+        } else if (type.equals("s3")) {
```
Comment (Vatavuk) [inline src/main/java/com/artipie/YamlSettings.java]: @olegmoz let's make it `"s3".equals(type)`, or reverse previous if statement to `type.equals("fs")` just to be consistent.

**Comment by olegmoz** [comment]
@Vatavuk made changes by your comments, please review

**Comment by 0crat** [comment]
Job was finished in 47 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR, performer @vatavuk[/z](https://www.0crat.com/u/vatavuk)[/z](https://www.0crat.com/u/vatavuk) didn't make CR comments

**Pull request merged**